tests/MarginCalculatorIntegrationTest.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+namespace PivotLibre\Tideman;
+
+use PivotLibre\Tideman\TestScenario1;
+use PivotLibre\Tideman\MarginCalculator;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
```

Comment:
This class is probably the only one worth reviewing. I'm particularly interested if you think the doc block describes how the tests were created with sufficient clarity.